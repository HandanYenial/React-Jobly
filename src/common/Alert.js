import React from "react";

/**Alert component to show bootstrap-style alerts
 * Used in Loginform , SignupForm, ProfileForm
 * 
 */

function Alert({type="danger" , messages = []}){ //default type is danger and messages is the array of error messages in other components
    console.debug(
        "Alert",
        "type=" ,type,
        "messages=" , messages);
        
        return(
            <div className = {`alert alert-${type}`} role = "alert">
                {messages.map(error => (
                    <p className = "mb-0 small" key = {error}>
                        {error}
                    </p>
                ))}
            </div>
        );
}

export default Alert;