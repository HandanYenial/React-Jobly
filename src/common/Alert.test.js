import React from "react";
import Alert from"./Alert";
import { render } from "@testing-library/react";

//Smoke test
it("renders without crashing" , function(){
    render(<Alert/>);
});

//Snapshot Test for danger message
it ("matches with the snapshot" , function(){
    let messages = ["danger", "Broken app"];
    const { asFragment } = render(<Alert type = "danger" messages={messages}/>);
    expect(asFragment()).toMatchSnapshot();
});

//Snapshot Test for success message
it("matches with the success snapshot", function(){
    let messages = ["success", "Awesome"];
    const { asFragment } = render(<Alert type = "success" messages={messages}/>);
    expect(asFragment()).toMatchSnapshot();
});

