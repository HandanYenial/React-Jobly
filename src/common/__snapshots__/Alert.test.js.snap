// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches with the snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger"
    role="alert"
  >
    <p
      class="mb-0 small"
    >
      danger
    </p>
    <p
      class="mb-0 small"
    >
      Broken app
    </p>
  </div>
</DocumentFragment>
`;

exports[`matches with the success snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
    role="alert"
  >
    <p
      class="mb-0 small"
    >
      success
    </p>
    <p
      class="mb-0 small"
    >
      Awesome
    </p>
  </div>
</DocumentFragment>
`;
